### PR TITLE
use pod_name in mobile meta description

### PR DESCRIPTION
**Pull request by ghost**
Use `pod_name` in the mobile meta description instead of `diaspora*`.

### CODE REVIEW

File: app/views/layouts/application.mobile.haml [hunk 1]
```diff
@@ -8,7 +8,7 @@
     %title
       = pod_name
 
-    %meta{:name => "description", :content => "diaspora* mobile"}/
+    %meta{:name => "description", :content => "#{pod_name} mobile"}/
```
Comment (diaspora-code-review) [inline app/views/layouts/application.mobile.haml]: Use the new Ruby 1.9 hash syntax.

Comment (svbergerem) [inline app/views/layouts/application.mobile.haml]: For the desktop view we do the following: 

https://github.com/diaspora/diaspora/blob/b319e0caa24c7387d1040668284bec747da08f9f/app/views/layouts/application.html.haml#L13
https://github.com/diaspora/diaspora/blob/b319e0caa24c7387d1040668284bec747da08f9f/app/helpers/meta_data_helper.rb#L29-L38

This uses [the description from the diaspora.yml file](https://github.com/diaspora/diaspora/blob/b319e0caa24c7387d1040668284bec747da08f9f/config/diaspora.yml.example#L557) which should be appropriate for both desktop and mobile views so I'd prefer some unification here.

File: app/views/layouts/application.html.haml [hunk 2]
```diff
@@ -3,24 +3,30 @@
 -#   the COPYRIGHT file.
 
 !!!
-%html{lang:  I18n.locale.to_s, dir:  (rtl?) ? 'rtl' : 'ltr'}
+%html{lang:  I18n.locale.to_s, dir:  (rtl?) ? "rtl" : "ltr"}
```
Comment (diaspora-code-review) [inline app/views/layouts/application.html.haml]: Don't use parentheses around a method call.

File: app/views/layouts/application.mobile.haml [hunk 3]
```diff
@@ -3,56 +3,52 @@
 -# the COPYRIGHT file.
 
 !!!
-%html{:lang => I18n.locale.to_s, :dir => (rtl?) ? 'rtl' : 'ltr'}
-  %head
+%html{:lang => I18n.locale.to_s, :dir => (rtl?) ? "rtl" : "ltr"}
```
Comment (diaspora-code-review) [inline app/views/layouts/application.mobile.haml]: - Use the new Ruby 1.9 hash syntax.
- Don't use parentheses around a method call.

File: app/views/layouts/application.mobile.haml [hunk 4]
```diff
@@ -3,56 +3,52 @@
...
+  %head{prefix: og_prefix}
     %title
-      = pod_name
+      = page_title yield(:page_title)
 
-    %meta{:name => "description", :content => "diaspora* mobile"}/
-    %meta{:name => "author", :content => "Diaspora, Inc."}/
-    %meta{:charset => 'utf-8'}/
+    %meta{charset:  "utf-8"}/
+    %meta{:name =>"viewport", :content => "width=device-width, minimum-scale=1 maximum-scale=1"}/
```
Comment (diaspora-code-review) [inline app/views/layouts/application.mobile.haml]: - Use the new Ruby 1.9 hash syntax.
- Surrounding space missing for operator `=>`.

File: app/views/layouts/application.mobile.haml [hunk 5]
```diff
@@ -3,56 +3,52 @@
...
     = include_color_theme "mobile"
-    = yield(:custom_css)
-
 
-    = csrf_meta_tag
+    = yield(:custom_css)
 
     - if rtl?
-      = stylesheet_link_tag :rtl, :media => 'all'
+      = stylesheet_link_tag :rtl, :media => "all"
```
Comment (diaspora-code-review) [inline app/views/layouts/application.mobile.haml]: Use the new Ruby 1.9 hash syntax.

File: app/views/layouts/application.mobile.haml [hunk 6]
```diff
@@ -3,56 +3,52 @@
...
-    / Viewport scale
-    %meta{:name =>'viewport', :content => "width=device-width, minimum-scale=1 maximum-scale=1"}/
-    %meta{:name => "HandheldFriendly", :content => "True"}/
-    %meta{:name => "MobileOptimized", :content => "320"}/
-    / Force cleartype on WP7
-    %meta{'http-equiv' => "cleartype", :content => 'on'}/
+    / mobile meta tags
+    %meta{name: "HandheldFriendly", content: "True"}/
+    %meta{name: "MobileOptimized", content: "320"}/
+    %meta{"http-equiv": "cleartype", content: "on"}/
```
Comment (diaspora-code-review) [inline app/views/layouts/application.mobile.haml]: unexpected token tCOLON
(Using Ruby 2.0 parser; configure using `TargetRubyVersion` parameter, under `AllCops`)

File: app/views/layouts/application.mobile.haml [hunk 7]
```diff
@@ -3,56 +3,52 @@
...
-    / Viewport scale
-    %meta{:name =>'viewport', :content => "width=device-width, minimum-scale=1 maximum-scale=1"}/
-    %meta{:name => "HandheldFriendly", :content => "True"}/
-    %meta{:name => "MobileOptimized", :content => "320"}/
-    / Force cleartype on WP7
-    %meta{'http-equiv' => "cleartype", :content => 'on'}/
+    / mobile meta tags
+    %meta{name: "HandheldFriendly", content: "True"}/
+    %meta{name: "MobileOptimized", content: "320"}/
+    %meta{"http-equiv" => "cleartype", content: "on"}/
```
Comment (diaspora-code-review) [inline app/views/layouts/application.mobile.haml]: Don't mix styles in the same hash.

File: app/views/layouts/application.html.haml [hunk 8]
```diff
@@ -3,24 +3,30 @@
...
+    %link{rel: "shortcut icon", href: image_path("apple-touch-icon.png")}
+    / All others
+    %link{rel: "shortcut icon", href: image_path("favicon.png")}
 
     = chartbeat_head_block
     = include_mixpanel
-
     = include_color_theme
 
+    = yield(:custom_css)
```
Comment (svbergerem) [inline app/views/layouts/application.html.haml]: Looks unused to me so we can drop this from both files.

**Review by svbergerem — CHANGES_REQUESTED** [review]
Wow, I didn't expect this huge amount of unification but it looks good to me. 👍 Please squash your commits into one after removing `yield(:custom_css)`. Then this should be ready to merge.

**Comment by jramos** [comment]
Will do. I toyed with the idea of a shared partial for the page head. WDYT?

**Comment by svbergerem** [comment]
Since there is a lot of shared code this seems legit. :)

**Comment by SuperTux88** [comment]
I'm unsure about the `is_mobile_device?`. Creating a shared partial where you need to if-else again feels wrong.

**Comment by Flaburgan** [comment]
+1 with superTux please extract the most mobile specific code you can to the mobile file.

File: app/views/layouts/application.mobile.haml [hunk 9]
```diff
@@ -3,56 +3,21 @@
...
 
 !!!
-%html{:lang => I18n.locale.to_s, :dir => (rtl?) ? 'rtl' : 'ltr'}
-  %head
-    %title
-      = pod_name
+%html{lang: I18n.locale.to_s, dir: (rtl? ? "rtl" : "ltr")}
+  %head{prefix: og_prefix}
+    = render "head"
+    = include_color_theme
```
Comment (svbergerem) [inline app/views/layouts/application.mobile.haml]: I guess this should be `include_color_theme "mobile"`.

**Review by svbergerem — APPROVED** [review]

**Comment by svbergerem** [comment]
Merged as 9aac486510a0b33a9f6a86b42347a62dbe6c95b1. Thanks!

**Comment by jramos** [comment]
@svbergerem -- actually, there's an issue with this. Any javascript in `content_for(:head)` will get executed before the `javascript_include_tag` calls happen, so `Diaspora` won't be defined. Working on a fix now....

**Pull request closed without merging**